File: Exemplos/App_IntentDigitalHUB_XamarinForms/Elgin_IntentDigitalHUB_XamarinForms-font/E1_Digital_Hub_Xamarin_Forms_M8/IntentServices/Bridge/IniciaVendaCredito.cs
namespace Xamarin_Forms_Intent_Digital_Hub.IntentServices.Bridge
{
    class IniciaVendaCredito : BridgeCommand
    {
        readonly int idTransacao;
        readonly string pdv;
        readonly string valorTotal;
        readonly int tipoFinanciamento;
        readonly int numeroParcelas;

        public IniciaVendaCredito(int idTransacao, string pdv, string valorTotal, int tipoFinanciamento, int numeroParcelas) : base("IniciaVendaCredito")
        {
            this.idTransacao = idTransacao;
            this.pdv = pdv;
            this.valorTotal = valorTotal;
            this.tipoFinanciamento = tipoFinanciamento;
            this.numeroParcelas = numeroParcelas;
        }

        protected override string FunctionParameters()
        {
            return "\"idTransacao\"" + ":" + this.idTransacao + "," +
                    "\"pdv\"" + ":" + "\"" + this.pdv + "\"" + "," +
                    "\"valorTotal\"" + ":" + "\"" + this.valorTotal + "\"" + "," +
                    "\"tipoFinanciamento\"" + ":" + this.tipoFinanciamento + "," +
                    "\"numeroParcelas\"" + ":" + this.numeroParcelas;
        }
    }
}

File: Exemplos M10/App_IntentDigitalHUB_ReactNative/Elgin_IntentDigitalHUB_ReactNative-font/src/intentDigitalHubService/BRIDGE/IniciaVendaCredito.ts
import {BridgeCommand} from './BridgeCommand';

export class IniciaVendaCredito extends BridgeCommand {
  readonly idTransacao: number;
  readonly pdv: string;
  readonly valorTotal: string;
  readonly tipoFinanciamento: number;
  readonly numeroParcelas: number;

  constructor(
    idTransacao: number,
    pdv: string,
    valorTotal: string,
    tipoFinanciamento: number,
    numeroParcelas: number,
  ) {
    super('IniciaVendaCredito');
    this.idTransacao = idTransacao;
    this.pdv = pdv;
    this.valorTotal = valorTotal;
    this.tipoFinanciamento = tipoFinanciamento;
    this.numeroParcelas = numeroParcelas;
  }

  functionParameters(): object {
    return {
      idTransacao: this.idTransacao,
      pdv: this.pdv,
      valorTotal: this.valorTotal,
      tipoFinanciamento: this.tipoFinanciamento,
      numeroParcelas: this.numeroParcelas,
    };
  }
}
